flows/flows.py

```diff
@@ -211,11 +214,15 @@ def encode_categorical_feature(self, dataframes_dict: dict, print_results: Union
 
         if not _reference:
             _reference = list(dataframes_dict.keys())[0]
```

Comment:
The Dictionary `keys()` function in Python returns a memory view and it's not a good idea to materialize it in memory as it will result in duplication of data, and that might be big(?). If you just want to access the first element, I propose the following alternative:

```
_reference = next(iter(dataframes_dict.keys()))
```

What it does is just create an iterator object on top of the memory view (`iter(my_dict.keys())`) and just consumes the first value (which is what the `next()` function does), this will save you the memory costs of saving the `keys()` output in a list in memory.

Reference: https://docs.python.org/3.7/library/stdtypes.html#dictionary-view-objects

Reply:
updated in all the methods inside the flow package